tests/stress/compute_flow_with_threads.py

```diff
@@ -261,17 +239,45 @@
     # Retrieve algorithm output and log files
     output = None
     for i in range(len(status["results"])):
-        result = None
         result_type = status["results"][i]["type"]
         print(f"Fetch result index {i}, type: {result_type}")
-        result = ocean.compute.result(DATA_asset, job_id, i, bob_wallet)
+        result = ocean.compute.result(DATA_asset, compute_service, job_id, i, wallet)
         assert result, "result retrieval unsuccessful"
+        print(f"result index: {i}, type: {result_type}, contents: {result}")
 
         # Extract algorithm output
         if result_type == "output":
             output = result
     assert output, "algorithm output not found"
+    unpickle_result(output)
+
 
-    # Unpickle the gaussian model result
-    model = pickle.loads(output)
+def unpickle_result(pickled):
+    """Unpickle the gaussian model result"""
+    model = pickle.loads(pickled)
     assert len(model) > 0, "unpickle result unsuccessful"
+
+
+def concurrent_c2d(concurrent_flows: int, duration: int):
+    config = ExampleConfig.get_config()
+    ocean = Ocean(config)
+    mint_fake_OCEAN(config)
+    with ThreadPoolExecutor(max_workers=concurrent_flows) as executor:
+        for _ in range(concurrent_flows * duration):
+            executor.submit(
+                c2d_flow_readme,
+                ocean,
+                "brainin",
+                "https://raw.githubusercontent.com/oceanprotocol/c2d-examples/main/branin_and_gpr/branin.arff",
+                "gpr",
+                "https://raw.githubusercontent.com/oceanprotocol/c2d-examples/main/branin_and_gpr/gpr.py",
+                "python-brain",
+            )
+
+
+@pytest.mark.slow
+def test_concurrent_c2d():
```

Comment:
I'd like to see this test parameterized so that all three concurrency settings are run (even if one of them fails)

```python
@pytest.mark.slow
@pytest.mark.parameterize(["concurrent_flows", "repetitions"], [(1, 3000), (3, 1000), (20, 50)])
def test_concurrent_c2d(concurrent_flows: int, repetitions: int)
    concurrent_c2d(concurrent_flows, repetitions)
```